Cron/CancelPendingOrders.php

```diff
@@ -0,0 +1,99 @@
+<?php
+namespace Razorpay\Magento\Cron;
+use \Magento\Sales\Model\Order;
+
+class CancelPendingOrders {
+    /**
+     * @var \Magento\Sales\Api\OrderRepositoryInterface
+     */
+    protected $orderRepository;
+    /**
+     * @var \Magento\Framework\Api\SearchCriteriaBuilder
+     */
+    protected $searchCriteriaBuilder;
+
+    /**
+     * @var \Magento\Sales\Api\OrderManagementInterface
+     */
+    protected $orderManagement;
+
+    /**
+     * CancelOrder constructor.
+     * @param \Magento\Sales\Api\OrderRepositoryInterface $orderRepository
+     * @param \Magento\Framework\Api\SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param \Magento\Framework\Api\SortOrderBuilder $sortOrderBuilder
+     * @param \Magento\Sales\Api\OrderManagementInterface $orderManagement
+     * @param \Razorpay\Magento\Model\Config $config
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(
+        \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
+        \Magento\Framework\Api\SearchCriteriaBuilder $searchCriteriaBuilder,
+        \Magento\Framework\Api\SortOrderBuilder $sortOrderBuilder,
+        \Magento\Sales\Api\OrderManagementInterface $orderManagement,
+        \Razorpay\Magento\Model\Config $config,
+        \Psr\Log\LoggerInterface $logger
+    )
+    {
+        $this->orderRepository                 = $orderRepository;
+        $this->searchCriteriaBuilder           = $searchCriteriaBuilder;
+        $this->sortOrderBuilder                = $sortOrderBuilder;
+        $this->orderManagement                 = $orderManagement;
+        $this->config                          = $config;
+        $this->logger                          = $logger;
+        $this->isCancelPendingOrderCronEnabled = $this->config->isCancelPendingOrderCronEnabled();
+        $this->pendingOrderTimeout             = ($this->config->getPendingOrderTimeout() > 0) ? $this->config->getPendingOrderTimeout() : 30;
+    }
+
+    public function execute()
+    {
+        // Execute only if Cancel Pending Order Cron is Enabled
+        if($this->isCancelPendingOrderCronEnabled === true && $this->pendingOrderTimeout > 0)
+        {
+            $this->logger->info("Cronjob: Cancel Pending Order Cron started.");
+            $dateTimeCheck = date('Y-m-d H:i:s',strtotime('-' . $this->pendingOrderTimeout . ' minutes'));
```

Comment:
space after comma

Reply:
Is it going to works fine in case of different time zones.

Reply:
@ramth05 Updated space after comma

Reply:
@ramth05 Magento system's default timezone is UTC, date function creates date with system's timezone. For creating an order and executing cron timezone must be same, so any timezone will work here. Updated timezone to Asia/Kolkata for testing changes.

Reply:
@enishant , please verify this field must be set through PHP script not directly in mysql.